packages/cli/src/ui/components/InputPrompt.tsx

```diff
@@ -58,10 +58,46 @@ export const InputPrompt: React.FC<InputPromptProps> = ({
   setShellModeActive,
 }) => {
   const [justNavigatedHistory, setJustNavigatedHistory] = useState(false);
+
+  // Check if cursor is after @ or / without spaces
+  const isCursorAfterCommandWithoutSpace = useCallback(() => {
+    const text = buffer.text;
+    const [row, col] = buffer.cursor;
+
+    // Calculate offset from row/col (same logic as in text-buffer.ts)
+    let offset = 0;
+    for (let i = 0; i < row; i++) {
+      offset += buffer.lines[i].length + 1; // +1 for newline
+    }
+    offset += col;
+
+    // Search backwards from cursor position
+    for (let i = offset - 1; i >= 0; i--) {
+      const char = text[i];
+      if (char === ' ' || char === '\n') {
```

Comment:
also need to handle escaped spaces in paths. check the existing logic for autocomplete for file paths to make sure the logic is aligned.